src/rgw/rgw_log.cc

```diff
@@ -200,21 +200,24 @@ static void log_usage(req_state *s, const string& op_name)
   if (!usage_logger)
     return;
 
-  std::string user;
+  std::string user = to_string(s->owner.id);
   std::string payer;
-  string bucket_name;
-
-  bucket_name = s->bucket_name;
+  string bucket_name = s->bucket_name;
 
   if (!bucket_name.empty()) {
-    bucket_name = s->bucket_name;
     user = to_string(s->bucket_owner.id);
+
+    // As per https://docs.aws.amazon.com/AmazonS3/latest/userguide/RequesterPaysBuckets.html#ChargeDetails
+    // If the bucket has requester pays enabled,
+    // and the requerster includes x-amz-request-payer in the header,
+    // and the status code isn't 403,
+    // then the requester is the payer.
     if (!rgw::sal::Bucket::empty(s->bucket.get()) &&
-	s->bucket->get_info().requester_pays) {
+        s->bucket->get_info().requester_pays &&
+        s->get_request_payer() &&
```

Comment:
I found that we're not checking for the header existence of `x-amz-request-payer` also, so I changed the commit in this regard.